### ISSUE TITLE
Chat - App fails to scroll chat to the first unread message in offline mode

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#42148]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.72-0
**Reproducible in staging?:** y 
**Reproducible in production?:** n
**Issue found when executing PR :** https://github.com/Expensify/App/pull/41448
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Navigate to staging.new.expensify.com
2. Start a chat with user B
3. Navigate to the LHN if on mWeb navigate to another chat if on web
4. Send multiple messages from user B to user A until there is a scrollable space meaning the new unread marker would be outside (higher up) the initial view range when the report is opened
5. Wait until the last message sent from user B is delivered to user A
6. Go offline once step 5 is completed
7. Open the chat with user B

## Expected Result:
App automatically scrolls to the first unread message

## Actual Result:
App fails to scroll user to the first unread message while in offline mode

## Workaround:
n/a

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/93399543/06e4c18b-260f-4895-a46e-c4e2643206c3

https://github.com/Expensify/App/assets/93399543/d9513293-65be-488c-bea8-65812198b127

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by kbecciv** [comment]
We think that this bug might be related to #wave-collect - Release 1

**Comment by Julesssss** [comment]
Very likely App instead of web.

I also think this is not an App blocker as this only occurs while offline. Lets get this fixed at a lower priority

**Comment by ikevin127** [comment]
Looking into this as we speak since I authored PR https://github.com/Expensify/App/pull/41448.
Will get back with a response regarding whether this was caused by my PR and if so I'll handle the follow-up.

**Comment by ikevin127** [comment]
I cannot reproduce this, it works for me with the given steps. I tried offline mode from Troubleshoot and also from browser Networking tab and both works for me on latest main (local dev) (see video below).

> [!note]
> It doesn't make much sense that this would be related to my PR as long as the out-of-view message is correctly marked as unread by BE before going offline.
>
> This is why I'm thinking that since this was tested on the staging server where we're currently experiencing BE issues with API's having delays, this could cause the new marker to not be set in the first place. See https://github.com/Expensify/App/issues/42063 for reference (which was also mistaken as a regression of my PR).

https://github.com/Expensify/App/assets/56457735/7e93aee4-470d-4f5f-86e7-4096210ceb98

**Comment by ikevin127** [comment]
Regardless, we decided to revert the PR -> more context @ https://github.com/Expensify/App/issues/35011#issuecomment-2110989063.

**Comment by Julesssss** [comment]
Thanks for explaining. We'll retest once the site is stable.

**Comment by ikevin127** [comment]
After thinking about what happens when navigating to a report while online vs offline, I confirm that this was indeed caused by the now-reverted PR and here's why:
1. When opening a report with new messages online, `OpenReport` API call is triggered.
2. Since I had `sortedVisibleReportActions` in the useEffect dependency array, when `OpenReport` succeeds -> `sortedVisibleReportActions` is updated -> re-running the effect -> scrolling to the new unread marker.
3. The logic was faulty since the effect was working based on (2.) logic.
4. While offline, the `OpenReport` API call is NOT triggered -> scroll doesn't happen while offline.

**Comment by ikevin127** [comment]
I'm working on the 2nd PR's functionality, will update again once PR is ready for review.

**Comment by ikevin127** [comment]
This should be closed since it was reported as regression from PR https://github.com/Expensify/App/pull/41448 which was reverted. So this shouldn't be an issue anymore since the functionality doesn't exist.
